cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/writer/LineageWriterDatasetFramework.java

```diff
@@ -91,12 +99,35 @@ public void deleteAllModules(Id.Namespace namespaceId) throws DatasetManagementE
   public void addInstance(String datasetTypeName, Id.DatasetInstance datasetInstanceId, DatasetProperties props)
     throws DatasetManagementException, IOException {
     delegate.addInstance(datasetTypeName, datasetInstanceId, props);
+    // add system metadata for user datasets only
+    if (!isUserDataset(datasetInstanceId)) {
+      return;
+    }
+    AbstractSystemMetadataWriter systemMetadataWriter =
+      new DatasetSystemMetadataWriter(metadataStore, dsInstantiatorFactory, datasetInstanceId, props, datasetTypeName);
+    systemMetadataWriter.write();
+  }
+
+  //TODO: Improve?
+  private boolean isUserDataset(Id.DatasetInstance datasetInstanceId) {
+    return !Id.Namespace.SYSTEM.equals(datasetInstanceId.getNamespace()) &&
+      !"system.queue.config".equals(datasetInstanceId.getId()) &&
+      !datasetInstanceId.getId().startsWith("system.sharded.queue") &&
```

Comment:
just wanted to be as strict as possible. Anyway, we need to make this easier/robust somehow - https://issues.cask.co/browse/CDAP-4627